osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/utils/DlmsHelper.java

```diff
@@ -198,7 +198,13 @@ public void checkResultList(
           e);
     } catch (final Exception e) {
       throw new ProtocolAdapterException(
-          "Error retrieving values with-list for device: " + device.getDeviceIdentification(), e);
+          "Error retrieving values with-list for device: "
+              + device.getDeviceIdentification()
+              + ", with-list: "
+              + (device.isWithListSupported() ? "supported" : "not supported")
+              + ", error: "
+              + (e.getMessage() != null ? e.getMessage() : "-"),
```

Comment:
This was added to give a bit more information about the cause of the error "Error retrieving values with-list for device" on the  headend, without the need to look in the logs.

Reply:
I am not sure that it is wrong here, but neither am I convinced it is the right thing to do.
You need to take into account different audiences with different needs or permissions to have access to certain information.
If I read it correctly this is to make sure more details are made available to users of a client application from the response.
I would say that they generally would only need (be allowed?) to know what went wrong up to the detail that was included in the error message. (There is no telling what the message of the caught `Exception` may contain.)
More detailed information is only for those who need to figure things out at a deeper technical level, which may be other people.

Reply:
Because we indeed don't know what the message of the exception might contain, I've removed it for now. Although I still think it would be good to try to give a bit more information on what went wrong (e.g. was it a connection problem or an error in the meter?)